src/Microsoft.AspNetCore.OData/Extensions/ODataRouteBuilderExtensions.cs

```diff
@@ -25,6 +25,382 @@ namespace Microsoft.AspNet.OData.Extensions
     /// </summary>
     public static class ODataRouteBuilderExtensions
     {
+        /// <summary>
+        /// Sets the <see cref="DefaultQuerySettings"/> in route builder.
+        /// </summary>
+        /// <param name="builder">The <see cref="IRouteBuilder"/>.</param>
+        /// <param name="defaultQuerySettings">The default query settings.</param>
+        public static IRouteBuilder SetDefaultQuerySettings(this IRouteBuilder builder, DefaultQuerySettings defaultQuerySettings)
+        {
+            if (builder == null)
+            {
+                throw Error.ArgumentNull("builder");
+            }
+
+            if (defaultQuerySettings == null)
+            {
+                throw Error.ArgumentNull("defaultQuerySettings");
+            }
+
+            if (!defaultQuerySettings.MaxTop.HasValue || defaultQuerySettings.MaxTop > 0)
+            {
+                ModelBoundQuerySettings.DefaultModelBoundQuerySettings.MaxTop = defaultQuerySettings.MaxTop;
+            }
+
+            DefaultQuerySettings querySettings = builder.ServiceProvider.GetRequiredService<DefaultQuerySettings>();
```

Comment:
It's "get" or "set"?

Reply:
Get existing object service then copy the values from the new object to the existing object.